cura/CuraPackageManager.py

```diff
@@ -31,13 +29,21 @@ def __init__(self, parent = None):
         self._plugin_registry = self._application.getPluginRegistry()
 
         # JSON file that keeps track of all installed packages.
-        self._package_management_file_path = os.path.join(os.path.abspath(Resources.getDataStoragePath()),
-                                                          "packages.json")
-        self._installed_package_dict = {}  # a dict of all installed packages
-        self._to_remove_package_set = set()  # a set of packages that need to be removed at the next start
-        self._to_install_package_dict = {}  # a dict of packages that need to be installed at the next start
+        self._bundled_package_management_file_path = os.path.join(
+            Application.getInstallPrefix(),
+            Application.getInstance().getApplicationName(),
+            "resources",
+            "packages.json"
+        )
+        self._user_package_management_file_path = os.path.join(os.path.abspath(Resources.getDataStoragePath()), "packages.json")
+
+
+        self._bundled_package_dict = {}     # A dict of all bundled packages
+        self._installed_package_dict = {}   # A dict of all installed packages
+        self._to_remove_package_set = set() # A set of packages that need to be removed at the next start
+        self._to_install_package_dict = {}  # A dict of packages that need to be installed at the next start
 
-    installedPackagesChanged = pyqtSignal()  # Emitted whenever the installed packages collection have been changed.
+    installedPackagesChanged = pyqtSignal() # Emitted whenever the installed packages collection have been changed.
 
     def initialize(self):
         self._loadManagementData()
```

Comment:
In _installAllScheduledPackages, we should probably add the installed packages to _installed_package_dict. Currently the packages.json in the share folder does not contain any "installed" packages.

Reply:
In _installAllScheduledPackages, I would see it as a kind of queue, install one, remove one, etc. Instead of installing everything, then emptying the list.

Reply:
Fixed in d6ad8ed